src/main/java/org/apache/commons/io/FilenameUtils.java

```diff
@@ -1490,4 +1496,163 @@ public static boolean wildcardMatch(final String fileName, final String wildcard
         return list.toArray( new String[ list.size() ] );
     }
 
+    /**
+     * Checks whether a given string is a valid host name according to
+     * RFC 3986.
+     *
+     * <p>Accepted are IP addresses (v4 and v6) as well as what the
+     * RFC calls a "reg-name". Percent encoded names don't seem to be
+     * valid names in UNC paths.</p>
+     *
+     * @see "https://tools.ietf.org/html/rfc3986#section-3.2.2"
+     * @param name the hostname to validate
+     * @return true if the given name is a valid host name
+     */
+    private static boolean isValidHostName(String name) {
+        return isIPv4Address(name) || isIPv6Address(name) || isRFC3986HostName(name);
+    }
+
+    private static final Pattern IPV4_PATTERN =
+        Pattern.compile("^(\\d{1,3})\\.(\\d{1,3})\\.(\\d{1,3})\\.(\\d{1,3})$");
+    private static final int IPV4_MAX_OCTET_VALUE = 255;
+
+    /**
+     * Checks whether a given string represents a valid IPv4 address.
+     *
```

Comment:
Does the code really need to check for more than just 4 sets of 1-3 digits? If a user enters 999.000.999.000 does that affect the final outcome?

Reply:
Lets concentrate on the aspects, that were causing this PR: IMO, it is perfectly fine to accept the IPV4_PATTERN as harmless, even if there is a possibility, that an error like INVALID_HOSTNAME, or whatever, may come up later.